tests/integration/test_agent.py

```diff
@@ -28,6 +28,17 @@
 print(f'workspace_mount_path_in_sandbox: {workspace_mount_path_in_sandbox}')
 
 
+def validate_final_state(final_state: State | None):
+    assert final_state is not None
+    assert final_state.agent_state == AgentState.STOPPED
```

Comment:
Just curious: why only `STOPPED` and not also `FINISHED`?

Reply:
In `main.py`, we close the agent controller before we return, thus it's `STOPPED` rather than `FINISHED`.

IIRC I used to return the last but one state in `main.py` so that it's `FINISHED`, but that change got reverted at some time.

Reply:
But in theory, even if it doesn't happen right now in main, `FINISHED` is a valid state and should be allowed or am I missing something here? 🤔

Reply:
If it's `FINISHED` it means the closing code breaks, so this assertion is kind of a mutation/change detector here.